src/core/thread/mle_router.cpp

```diff
@@ -447,14 +447,38 @@ void MleRouter::HandleAdvertiseTrickleTimer(void)
 
 void MleRouter::StopAdvertiseTrickleTimer(void) { mAdvertiseTrickleTimer.Stop(); }
 
+void MleRouter::RecalculateAdvertiseInterval(void)
+{
+    uint32_t neighbors;
+    uint32_t advertiseIntervalMax;
+
+    VerifyOrExit(IsRouterOrLeader());
+
+    neighbors            = Get<RouterTable>().GetNeighborCount();
+    advertiseIntervalMax = Min(kAdvertiseIntervalMax, (neighbors + 1) * kNeighborAdvertiseIntervalMultiplier);
```

Comment:
Can use `Get<RouterTable>().GetNeighborCount()` directly here (and not define a local `neighbors` variable)?

Reply:
This would also avoid assigning the uint8 return value into the uint32 first, which isn't needed.
Note that there was discussion on only using the Router neighbors that have  Link Quality >= 2 for this count.